python/flink_agents/api/tools/mcp.py

```diff
@@ -0,0 +1,132 @@
+
+        Default implementation is not provided; subclasses must override.
+        """
+        raise NotImplementedError(
+            "to_model_tool must be implemented by MCPToolDefinition subclasses"
+        )
+
+
+class MCPPrompt(BaseModel):
+    """MCP prompt definition returned by list_prompts/get_prompt."""
+
+    name: str
+    description: Optional[str] = None
+    arguments: Dict[str, Any] = Field(default_factory=dict)
+
+
+class MCPServer(BaseTool, ABC):
+    """Resource representing an MCP server and exposing its tools/prompts.
+
+    This is a logical tool container; it is not directly invokable with call().
+    """
+
+    endpoint: str
+    metadata: ToolMetadata = Field(  # default metadata for the server wrapper
+        default_factory=lambda: ToolMetadata(
+            name="mcp_server",
+            description="A container for tools provided by an MCP server.",
+            args_schema=BaseModel,  # not used
+        )
+    )
+
+    @classmethod
+    @override
+    def resource_type(cls) -> ResourceType:  # type: ignore[override]
+        return ResourceType.MCP_SERVER
+
+    @classmethod
+    @override
+    def tool_type(cls) -> ToolType:
+        return ToolType.MCP
+
+    def call_tool(self, name: str,*args: Any, **kwargs: Any) -> Any:  # noqa: D401
+        """MCPServer isn't directly invokable; raise to signal misuse."""
+        raise NotImplementedError("MCPServer does not support direct call().")
+
+    # Listing and retrieval APIs
+    def list_tools(self, cursor: Optional[str] = None) -> Tuple[List[MCPToolDefinition], Optional[str]]:
+        """List available tool definitions from the MCP server and an optional cursor.
+
+        Parameters
+        ----------
+        cursor : Optional[str]
+            Optional pagination cursor/state.
+
+        Returns
+        -------
+        Tuple[List[MCPToolDefinition], Optional[str]]
+            A tuple of (tools, next_cursor). next_cursor may be None when there are no more pages.
+        """
+        raise NotImplementedError
+
+    def get_tool_definition(self, name: str) -> MCPToolDefinition:
+        """Get a single tool definition by name."""
+        raise NotImplementedError
+
+    def list_prompts(self, cursor: Optional[str] = None) -> Tuple[List[MCPPrompt], Optional[str]]:
+        """List available prompts from the MCP server and an optional cursor.
+
+        Parameters
+        ----------
+        cursor : Optional[str]
+            Optional pagination cursor/state.
+
+        Returns
+        -------
+        Tuple[List[MCPPrompt], Optional[str]]
+            A tuple of (prompts, next_cursor). next_cursor may be None when there are no more pages.
+        """
+        raise NotImplementedError
+
+    def get_prompt(self, name: str) -> MCPPrompt:
+        """Get a single prompt definition by name."""
+        raise NotImplementedError
+
+    # Conversion helpers
+    def to_model_tool(self, name: str, chat_model: BaseChatModel) -> Dict[str, Any]:
```

Comment:
Passing mcp directly as tool to model instead of passing its individual tools to model